src/core/border_router/routing_manager.cpp

```diff
@@ -253,8 +253,6 @@ void RoutingManager::RecvIcmp6Message(uint32_t            aInfraIfIndex,
         HandleRouterAdvertisement(aSrcAddress, aBuffer, aBufferLength);
         break;
     case Ip6::Icmp::Header::kTypeRouterSolicit:
-        // Drop Router Solicitations initiated from infra interface.
-        VerifyOrExit(!otPlatInfraIfHasAddress(mInfraIfIndex, &aSrcAddress), error = kErrorDrop);
         HandleRouterSolicit(aSrcAddress, aBuffer, aBufferLength);
```

Comment:
Accepts RS messages initiated from the infra interface: this is for case that multiple Border Routers run on the same host (as we are trying to support multiple `ot-daemon`s on the same host).

Notice that we are already dropping the RS messages in `HandleRouterSolicit` when we are discovering routers:
https://github.com/openthread/openthread/blob/2796615cf7e64bc0fc633b6f8c6430dade1b466d/src/core/border_router/routing_manager.cpp#L892

So we will not respond to RS messages sent by ourselves.